include/cinder/audio/Target.h

```diff
@@ -32,30 +32,42 @@ namespace cinder { namespace audio {
 
 typedef std::shared_ptr<class TargetFile>		TargetFileRef;
 
+namespace dsp {
+	class Converter;
+}
+
 //! Base class that is used to create and write to an audio destination. Currently only supports .wav encoding.
 class CI_API TargetFile {
   public:
-	static std::unique_ptr<TargetFile> create( const DataTargetRef &dataTarget, size_t sampleRate, size_t numChannels, SampleType sampleType = SampleType::INT_16, const std::string &extension = "" );
-	static std::unique_ptr<TargetFile> create( const fs::path &path, size_t sampleRate, size_t numChannels, SampleType sampleType = SampleType::INT_16, const std::string &extension = "" );
-	virtual ~TargetFile() {}
+	static std::unique_ptr<TargetFile> create( const DataTargetRef &dataTarget, size_t sampleRate, size_t numChannels, SampleType sampleType = SampleType::INT_16, size_t targetSampleRate = 0, const std::string &extension = "" );
+	static std::unique_ptr<TargetFile> create( const fs::path &path, size_t sampleRate, size_t numChannels, SampleType sampleType = SampleType::INT_16, size_t targetSampleRate = 0, const std::string &extension = "" );
+	virtual ~TargetFile();
 
 	void write( const Buffer *buffer );
 	void write( const Buffer *buffer, size_t numFrames );
 	void write( const Buffer *buffer, size_t numFrames, size_t frameOffset );
 
+	//! Returns the user facing sample rate (input)
 	size_t getSampleRate() const	{ return mSampleRate; }
+	//! Returns the true sample rate of the target file. \note Actual input samplerate may differ. \see getSampleRate()
+	size_t getSampleRateTarget() const	{ return mSampleRateTarget; }
```

Comment:
For me both seem reasonable. I personally didn't find ``getSampleRateTarget()`` ambiguous since I thought of it as "get _the_ sample rate _of the_ target [file]". But I can definitely see the consistency argument with ``SourceFile``.